packages/tpetra/core/example/BlockCrs/driver.cpp

```diff
@@ -0,0 +1,536 @@
+// @HEADER
+// ***********************************************************************
+//
+//          Tpetra: Templated Linear Algebra Services Package
+//                 Copyright (2008) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+#include <cmath>
+#include <iostream>
+#include <iomanip>
+#include <sstream>
+
+#include <Tpetra_DefaultPlatform.hpp>
+#include <Tpetra_Version.hpp>
+
+#include <MatrixMarket_Tpetra.hpp>
+
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_GlobalMPISession.hpp>
+#include <Teuchos_FancyOStream.hpp>
+
+#include "typedefs.hpp"
+#include "utils.hpp"
+#include "MeshDatabase.hpp"
+
+//#define PRINT_VERBOSE 1
+
+using namespace BlockCrsTest;
+
+int main (int argc, char *argv[]) 
+{
+  using Teuchos::RCP;
+  using Teuchos::TimeMonitor;
+
+  const auto INVALID_GLOBAL_ORDINAL = Teuchos::OrdinalTraits<global_ordinal_type>::invalid();
+
+  auto out = Teuchos::getFancyOStream (Teuchos::rcpFromRef (std::cout));
+  
+  // MPI boilerplate
```

Comment:
Consider using Tpetra::initialize and Tpetra::finalize instead.  This takes care of both MPI and Kokkos initialization.

Reply:
fixed.